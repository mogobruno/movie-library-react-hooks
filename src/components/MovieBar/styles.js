export default theme => ({
  content: {
    width: '100%',
    backgroundColor: '#20232a',
    padding: '10px 30px'
  },
  title: {
    fontSize: '20px',
    color: '#ffffff'
  },
  movies: {
    padding: '20px',
    [theme.breakpoints.down('xs')]: {
      padding: '5px',
    },
    overflow: 'auto',
    display: 'flex'
  },
  moviesWrap: {
    [theme.breakpoints.down('xs')]: {
      padding: '5px',
    },
    display: 'flex',
    flexWrap: 'wrap',
    overflow: 'auto',
    height: '100%',
    justifyContent: 'space-between'
  },
})
