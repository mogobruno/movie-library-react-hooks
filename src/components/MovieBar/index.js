import React from 'react'
import PropTypes from 'prop-types'
import { withStyles } from '@material-ui/core/styles'

// Custom Hooks
import useMovieApi from 'hooks/useMovieApi'

// Components
import MovieCard from 'components/MovieCard'

// Styles
import styles from './styles'

const MovieBar = ({ classes, title, request, wrap = false }) => {
  const [movies] = useMovieApi('get', request.path, request.query, request.body)

  if (!movies.results.length) {
    return null
  }

  return (
    <div className={classes.content}>
      <h1 className={classes.title}>{title}</h1>
      <div className={ wrap ? classes.moviesWrap : classes.movies }>
        {movies.results.map((movie) => {
          return <MovieCard movie={movie} key={movie.id} />
        })}
      </div>
    </div>
  )
}

MovieBar.propTypes = {
  classes: PropTypes.object.isRequired,
  request: PropTypes.object,
  title: PropTypes.string,
  wrap: PropTypes.bool
}

export default withStyles(styles)(MovieBar)
