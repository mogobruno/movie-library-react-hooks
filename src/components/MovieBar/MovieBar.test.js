import React from 'react'
import axios from 'axios'
import { mount } from 'enzyme'
import renderer from 'react-test-renderer'

import MovieBar from './index'

import mockMovies from './__mocks__/movies'

jest.mock('hooks/useMovieApi', () => {
  return () => [
    mockMovies,
    () => {}
  ]
})

describe('<MovieBar />', () => {

  let component

  beforeEach(() => {
    component = (
      <MovieBar
        wrap
        title="test title"
        request={{
          path: `/3/search/movie`,
          query: {
            query: 'test expression'
          }
        }
      } />
    )
  })

  test('toMatchSnapshot', () => {
    const treeEmptyList = renderer
      .create(component)
      .toJSON()
    expect(treeEmptyList).toMatchSnapshot()
  })
})
