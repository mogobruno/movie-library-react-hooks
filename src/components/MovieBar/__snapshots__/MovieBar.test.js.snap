// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MovieBar /> toMatchSnapshot 1`] = `
<div
  className="MovieBar-content-1"
>
  <h1
    className="MovieBar-title-2"
  >
    test title
  </h1>
  <div
    className="MovieBar-moviesWrap-4"
  >
    <div
      className="MovieCard-content-5"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <div
        className="MovieCard-vote-6"
      >
        <div
          className="Vote-content-12"
        >
          <span
            className="Vote-content-12"
          >
            7.8
          </span>
          <canvas
            className="Vote-canvas-13"
          />
        </div>
      </div>
      <img
        alt="How to Train Your Dragon: The Hidden World"
        className="MovieCard-image-7"
        src="https://image.tmdb.org/t/p/w500/h3KN24PrOheHVYs9ypuOIdFBEpX.jpg"
      />
    </div>
    <div
      className="MovieCard-content-5"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <div
        className="MovieCard-vote-6"
      >
        <div
          className="Vote-content-12"
        >
          <span
            className="Vote-content-12"
          >
            7.3
          </span>
          <canvas
            className="Vote-canvas-13"
          />
        </div>
      </div>
      <img
        alt="Captain Marvel"
        className="MovieCard-image-7"
        src="https://image.tmdb.org/t/p/w500/w2PMyoyLU22YvrGK3smVM9fW1jj.jpg"
      />
    </div>
    <div
      className="MovieCard-content-5"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <div
        className="MovieCard-vote-6"
      >
        <div
          className="Vote-content-12"
        >
          <span
            className="Vote-content-12"
          >
            6.5
          </span>
          <canvas
            className="Vote-canvas-13"
          />
        </div>
      </div>
      <img
        alt="Bumblebee"
        className="MovieCard-image-7"
        src="https://image.tmdb.org/t/p/w500/wtZj5nn6hVwgakPdg6y6gm3eFXU.jpg"
      />
    </div>
    <div
      className="MovieCard-content-5"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <div
        className="MovieCard-vote-6"
      >
        <div
          className="Vote-content-12"
        >
          <span
            className="Vote-content-12"
          >
            6.5
          </span>
          <canvas
            className="Vote-canvas-13"
          />
        </div>
      </div>
      <img
        alt="The Mule"
        className="MovieCard-image-7"
        src="https://image.tmdb.org/t/p/w500/bTeRgkAavyw1eCtSkaww18wLYNP.jpg"
      />
    </div>
    <div
      className="MovieCard-content-5"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <div
        className="MovieCard-vote-6"
      >
        <div
          className="Vote-content-12"
        >
          <span
            className="Vote-content-12"
          >
            6.3
          </span>
          <canvas
            className="Vote-canvas-13"
          />
        </div>
      </div>
      <img
        alt="Triple Frontier"
        className="MovieCard-image-7"
        src="https://image.tmdb.org/t/p/w500/s9I2LmQMYCanl6DvC3X1AOHs2r8.jpg"
      />
    </div>
    <div
      className="MovieCard-content-5"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <div
        className="MovieCard-vote-6"
      >
        <div
          className="Vote-content-12"
        >
          <span
            className="Vote-content-12"
          >
            5.4
          </span>
          <canvas
            className="Vote-canvas-13"
          />
        </div>
      </div>
      <img
        alt="葉問外傳：張天志"
        className="MovieCard-image-7"
        src="https://image.tmdb.org/t/p/w500/grtVFGJ4ts0nDAPpc1JWbBoVKTu.jpg"
      />
    </div>
    <div
      className="MovieCard-content-5"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <div
        className="MovieCard-vote-6"
      >
        <div
          className="Vote-content-12"
        >
          <span
            className="Vote-content-12"
          >
            6.1
          </span>
          <canvas
            className="Vote-canvas-13"
          />
        </div>
      </div>
      <img
        alt="Happy Death Day 2U"
        className="MovieCard-image-7"
        src="https://image.tmdb.org/t/p/w500/dhNJHBDacrZjSPtwaiwp3idpzxU.jpg"
      />
    </div>
    <div
      className="MovieCard-content-5"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <div
        className="MovieCard-vote-6"
      >
        <div
          className="Vote-content-12"
        >
          <span
            className="Vote-content-12"
          >
            6.7
          </span>
          <canvas
            className="Vote-canvas-13"
          />
        </div>
      </div>
      <img
        alt="Alita: Battle Angel"
        className="MovieCard-image-7"
        src="https://image.tmdb.org/t/p/w500/aQXTw3wIWuFMy0beXRiZ1xVKtcf.jpg"
      />
    </div>
    <div
      className="MovieCard-content-5"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <div
        className="MovieCard-vote-6"
      >
        <div
          className="Vote-content-12"
        >
          <span
            className="Vote-content-12"
          >
            7.2
          </span>
          <canvas
            className="Vote-canvas-13"
          />
        </div>
      </div>
      <img
        alt="Us"
        className="MovieCard-image-7"
        src="https://image.tmdb.org/t/p/w500/jNUCddkM1fjYcFIcEwFjc7s2H4V.jpg"
      />
    </div>
    <div
      className="MovieCard-content-5"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <div
        className="MovieCard-vote-6"
      >
        <div
          className="Vote-content-12"
        >
          <span
            className="Vote-content-12"
          >
            6
          </span>
          <canvas
            className="Vote-canvas-13"
          />
        </div>
      </div>
      <img
        alt="A Madea Family Funeral"
        className="MovieCard-image-7"
        src="https://image.tmdb.org/t/p/w500/piGjUloiaq6qTpaDLisyDyEsx9i.jpg"
      />
    </div>
    <div
      className="MovieCard-content-5"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <div
        className="MovieCard-vote-6"
      >
        <div
          className="Vote-content-12"
        >
          <span
            className="Vote-content-12"
          >
            4.4
          </span>
          <canvas
            className="Vote-canvas-13"
          />
        </div>
      </div>
      <img
        alt="Captive State"
        className="MovieCard-image-7"
        src="https://image.tmdb.org/t/p/w500/6IZ84KFhNpy6htsJmwFVuK30pyp.jpg"
      />
    </div>
    <div
      className="MovieCard-content-5"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <div
        className="MovieCard-vote-6"
      >
        <div
          className="Vote-content-12"
        >
          <span
            className="Vote-content-12"
          >
            6.7
          </span>
          <canvas
            className="Vote-canvas-13"
          />
        </div>
      </div>
      <img
        alt="Mary Poppins Returns"
        className="MovieCard-image-7"
        src="https://image.tmdb.org/t/p/w500/cwiJQXezWz876K3jS57Sq56RYCZ.jpg"
      />
    </div>
    <div
      className="MovieCard-content-5"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <div
        className="MovieCard-vote-6"
      >
        <div
          className="Vote-content-12"
        >
          <span
            className="Vote-content-12"
          >
            8.1
          </span>
          <canvas
            className="Vote-canvas-13"
          />
        </div>
      </div>
      <img
        alt="Bohemian Rhapsody"
        className="MovieCard-image-7"
        src="https://image.tmdb.org/t/p/w500/xcaSYLBhmDzJ6P14bcKe0KTh3QV.jpg"
      />
    </div>
    <div
      className="MovieCard-content-5"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <div
        className="MovieCard-vote-6"
      >
        <div
          className="Vote-content-12"
        >
          <span
            className="Vote-content-12"
          >
            5.2
          </span>
          <canvas
            className="Vote-canvas-13"
          />
        </div>
      </div>
      <img
        alt="Wonder Park"
        className="MovieCard-image-7"
        src="https://image.tmdb.org/t/p/w500/nJyAdKPnW15IAvC6sLBTE0lp6Dv.jpg"
      />
    </div>
    <div
      className="MovieCard-content-5"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <div
        className="MovieCard-vote-6"
      >
        <div
          className="Vote-content-12"
        >
          <span
            className="Vote-content-12"
          >
            8.3
          </span>
          <canvas
            className="Vote-canvas-13"
          />
        </div>
      </div>
      <img
        alt="Green Book"
        className="MovieCard-image-7"
        src="https://image.tmdb.org/t/p/w500/78PjwaykLY2QqhMfWRDvmfbC6EV.jpg"
      />
    </div>
    <div
      className="MovieCard-content-5"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <div
        className="MovieCard-vote-6"
      >
        <div
          className="Vote-content-12"
        >
          <span
            className="Vote-content-12"
          >
            4.2
          </span>
          <canvas
            className="Vote-canvas-13"
          />
        </div>
      </div>
      <img
        alt="Level 16"
        className="MovieCard-image-7"
        src="https://image.tmdb.org/t/p/w500/dSdZ9iAgoDyAh3AcXtdyS9GtBKu.jpg"
      />
    </div>
    <div
      className="MovieCard-content-5"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <div
        className="MovieCard-vote-6"
      >
        <div
          className="Vote-content-12"
        >
          <span
            className="Vote-content-12"
          >
            5.4
          </span>
          <canvas
            className="Vote-canvas-13"
          />
        </div>
      </div>
      <img
        alt="What Men Want"
        className="MovieCard-image-7"
        src="https://image.tmdb.org/t/p/w500/umecegsPpKr2ZXA62Da9CQBVoIO.jpg"
      />
    </div>
    <div
      className="MovieCard-content-5"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <div
        className="MovieCard-vote-6"
      >
        <div
          className="Vote-content-12"
        >
          <span
            className="Vote-content-12"
          >
            6.1
          </span>
          <canvas
            className="Vote-canvas-13"
          />
        </div>
      </div>
      <img
        alt="Fighting with My Family"
        className="MovieCard-image-7"
        src="https://image.tmdb.org/t/p/w500/mYKP6qcDUimVMAHQWLOuDHjO19S.jpg"
      />
    </div>
    <div
      className="MovieCard-content-5"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <div
        className="MovieCard-vote-6"
      >
        <div
          className="Vote-content-12"
        >
          <span
            className="Vote-content-12"
          >
            7.7
          </span>
          <canvas
            className="Vote-canvas-13"
          />
        </div>
      </div>
      <img
        alt="Shazam!"
        className="MovieCard-image-7"
        src="https://image.tmdb.org/t/p/w500/nmV9MdbzST4xv8WMHwhVgxkHHjM.jpg"
      />
    </div>
    <div
      className="MovieCard-content-5"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <div
        className="MovieCard-vote-6"
      >
        <div
          className="Vote-content-12"
        >
          <span
            className="Vote-content-12"
          >
            7.3
          </span>
          <canvas
            className="Vote-canvas-13"
          />
        </div>
      </div>
      <img
        alt="Ralph Breaks the Internet"
        className="MovieCard-image-7"
        src="https://image.tmdb.org/t/p/w500/88poTBTafMXaz73vYi3c74g0y2k.jpg"
      />
    </div>
  </div>
</div>
`;
